src/scheduler.rs

```diff
@@ -262,17 +262,18 @@ fn temporary_channel_id() -> ChannelId { rand::random() }
 pub struct Scheduler {
     lnd: LndClient,
     endpoint: Url,
+    lsp_endpoint: String,
     pjs: Arc<Mutex<HashMap<Script, ScheduledPayJoin>>>, // payjoins mapped by owned `scriptPubKey`s
 }
 
 impl Scheduler {
     /// New [Scheduler].
-    pub fn new(lnd: LndClient, endpoint: Url) -> Self {
-        Self { lnd, endpoint, pjs: Default::default() }
+    pub fn new(lnd: LndClient, endpoint: Url, lsp_endpoint: String) -> Self {
+        Self { lnd, endpoint, lsp_endpoint, pjs: Default::default() }
     }
 
     pub async fn from_config(config: &crate::config::Config) -> Result<Self, SchedulerError> {
-        Ok(Scheduler::new(LndClient::from_config(&config).await?, config.endpoint.parse().expect("Malformed secure endpoint from config file. Expecting a https or .onion URI to proxy payjoin requests")))
+        Ok(Scheduler::new(LndClient::from_config(&config).await?, config.endpoint.parse().expect("Malformed secure endpoint from config file. Expecting a https or .onion URI to proxy payjoin requests"), config.endpoint.parse().expect("Malformed secure lsp endpoint in config file. Expecting a https url")))
```

Comment:
I also don't think Scheduler should know about lsp_endpoint. Instead it could know about LspClient and its public interface including request_quote. Initializing LspClient `from_config would be a better encapsulated design.

Reply:
It should also be a `Url` type, like endpoint.